SignalMessaging/utils/ConversationSearcher.swift

```diff
@@ -75,30 +111,44 @@ public class ConversationSearcher: NSObject {
             if let thread = match as? TSThread {
                 let threadViewModel = ThreadViewModel(thread: thread, transaction: transaction)
                 let snippet: String? = thread.lastMessageText(transaction: transaction)
-                let searchResult = ConversationSearchResult(thread: threadViewModel, messageId: nil, snippet: snippet)
+                let sortKey = NSDate.ows_millisecondsSince1970(for: threadViewModel.lastMessageDate)
+                let searchResult = ConversationSearchResult(thread: threadViewModel, messageId: nil, snippet: snippet, sortKey: sortKey)
 
                 if let contactThread = thread as? TSContactThread {
                     let recipientId = contactThread.contactIdentifier()
                     existingConversationRecipientIds.insert(recipientId)
                 }
+
                 conversations.append(searchResult)
             } else if let message = match as? TSMessage {
                 let thread = message.thread(with: transaction)
 
                 let threadViewModel = ThreadViewModel(thread: thread, transaction: transaction)
-                let searchResult = ConversationSearchResult(thread: threadViewModel, messageId: message.uniqueId, snippet: snippet)
-
+                let sortKey = message.timestamp
+                let searchResult = ConversationSearchResult(thread: threadViewModel, messageId: message.uniqueId, snippet: snippet, sortKey: sortKey)
                 messages.append(searchResult)
             } else if let signalAccount = match as? SignalAccount {
-                let searchResult = ContactSearchResult(signalAccount: signalAccount)
+                let anyFont = UIFont.systemFont(ofSize: 12)
+                let sortKey = contactsManager.attributedStringForConversationTitle(withPhoneIdentifier: signalAccount.recipientId,
+                    primaryFont: anyFont,
+                    secondaryFont: anyFont).string
+
+                let searchResult = ContactSearchResult(signalAccount: signalAccount, sortKey: sortKey)
                 contacts.append(searchResult)
             } else {
                 owsFail("\(self.logTag) in \(#function) unhandled item: \(match)")
             }
         }
 
         // Only show contacts which were not included in an existing 1:1 conversation.
-        let otherContacts: [ContactSearchResult] = contacts.filter { !existingConversationRecipientIds.contains($0.recipientId) }
+        var otherContacts: [ContactSearchResult] = contacts.filter { !existingConversationRecipientIds.contains($0.recipientId) }
+
+        // Order the conversation and message results in reverse chronological order.
+        // The contact results are pre-sorted by display name.
+        conversations = conversations.sorted().reversed()
```

Comment:
Instead of enumerating the list twice, let's do `sorted(by: >)`

Reply:
Done.